src/backend/base/langflow/components/tools/currency_convertor.py

```
@@ -0,0 +1,130 @@
+            "access_key": self.api_key,
+            "source": self.base_currency,
+            "currencies": self.target_currencies,
+            "format": 1,
+        }
+        if self.mode == CurrencyMode.LIVE:
+            url = "http://api.exchangerate.host/live"
+        else:
+            url = "http://api.exchangerate.host/historical"
+            params["date"] = self.date
+
+        resp = requests.get(url, params=params)
```

Ruff (0.11.9)

Probable use of `requests` call without timeout

(S113)